inductiva/simulators/simulator.py

```diff
@@ -36,6 +39,15 @@ def _setup_input_dir(self, input_dir: types.Path):
                 f"The provided path (\"{input_dir}\") is not a directory.")
         return input_dir
 
+    def _check_resource_pool(self, machine_group):
+        if not self.checked_resource_pool:
```

Comment:
I suggest the reverse, name it `__check_active_resources` and set it to True first. Moreover, this argument needs to be hidden so users don't mess with it.